File: frontend/src/types/project.ts
export interface Project {
  _id: string
  organization_id: string
  workspace_id: string | null
  client_id: string | null
  name: string
  description: string
  industry: string
  project_type: string
  status: string
  proposal_ids: string[]
  goal?: string
  budget?: number | null
  currency?: string
  timeline?: string | null
  key_features?: string[]
  notes?: string
  created_by: string
  created_at: string
  updated_at: string
}

export interface ProjectCreateRequest {
  name: string
  description?: string
  industry?: string
  project_type?: string
  workspace_id?: string | null
  client_id?: string | null
}

export interface ProjectUpdateRequest {
  name?: string
  description?: string
  industry?: string
  project_type?: string
  workspace_id?: string | null
  client_id?: string | null
}

export interface HubProposal {
  _id: string
  proposal_id: string
  title: string
  status: string
  error: string | null
  created_at: string | null
}

export interface HubActivityEvent {
  _id: string
  event_type: string
  resource_type: string
  resource_id: string
  occurred_at: string
  payload?: Record<string, unknown> | null
}

export interface ProjectHub {
  project: Project
  proposals: HubProposal[]
  activity: HubActivityEvent[]
}

export interface ProjectHubUpdate {
  goal?: string
  budget?: number | null
  currency?: string
  timeline?: string | null
  key_features?: string[]
  notes?: string
  status?: string
}

File: backend/app/domain/entities/project.py
from datetime import datetime, timezone
from typing import Optional
from pydantic import BaseModel, Field, field_validator

from app.config.constants import IndustryType


class Project(BaseModel):
    id: Optional[str] = Field(default=None, alias="_id")

    @field_validator("id", mode="before")
    @classmethod
    def coerce_id(cls, v):
        return str(v) if v is not None else v
    organization_id: str
    workspace_id: Optional[str] = None
    client_id: Optional[str] = None
    name: str
    description: str = ""
    industry: IndustryType = IndustryType.CUSTOM
    project_type: str = "custom"
    status: str = "draft"
    proposal_ids: list[str] = Field(default_factory=list)
    goal: str = ""
    budget: Optional[float] = None
    currency: str = "USD"
    timeline: Optional[str] = None
    key_features: list[str] = Field(default_factory=list)
    notes: str = ""
    created_by: str
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    model_config = {"populate_by_name": True, "arbitrary_types_allowed": True}
